cluster-autoscaler/simulator/scheduler_based_predicates_checker.go

```diff
@@ -106,7 +107,8 @@ func (p *SchedulerBasedPredicateChecker) FitsAnyNodeMatching(clusterSnapshot Clu
 		return "", fmt.Errorf("error running pre filter plugins for pod %s; %s", pod.Name, preFilterStatus.Message())
 	}
 
-	for _, nodeInfo := range nodeInfosList {
+	for i := range nodeInfosList {
+		nodeInfo := nodeInfosList[(p.lastIndex+i)%len(nodeInfosList)]
```

Comment:
I cannot see any random anymore. Where did it go :) ?

Reply:
Good point, this approach is no longer based on the randomness, as it is replaced by deterministic round-robin. I am gonna change the commit name to better reflect that.

Reply:
The purpose of both the previous approach and round-robin is to ensure, that in the consecutive function calls pods are assigned to destination nodes somewhat evenly, so the group of chosen destination nodes is as big as possible - this ensures scale down works better.

Round-robin is better in terms of time complexity and repetitive memory allocation than the previous approach - shuffling nodes in each function call takes additional time and memory, while keeping round-robin index is relatively cost free. 

Other than that both these approaches are expected to work similarly - the difference is that in case of randomisation we rely on expected outcome while in case of round-robin we have more control over it.

Reply:
It's additionally worth to mention that I tested both proposed solutions, round-robin and shuffling of nodes, and found that the latter may have slightly negative impact on scale-up speed, as scale-up uses this function as well. Implementing the shuffling correctly would then require some additional code refactoring, which further confirmed my conviction to round-robin approach.